app/assets/javascripts/pages/settings.coffee

```diff
@@ -0,0 +1,35 @@
+PubSub.subscribe('assigned.website', (ev, website)->
+  switch gon.action
+    when "index"
+      $.getJSON Routes.api_v1_website_path(website.id), (data) ->
+        $('#current-website').render data
+        manipulateSettingsData(data)
+)
+
+manipulateSettingsData = (data) ->
+  $.website_id = data.id
+  $('#generateAppKey, #revoke').on 'click', (e)->
+    e.preventDefault();
+    $.randomKey = randomString()
+    while jQuery.inArray($.randomKey, gon.keys) >= 0
+      $.randomKey = randomString()
+    $('.app_key').html $.randomKey
+    $.ajax
+      data: {website: {id: $.website_id, app_key: $.randomKey}}
+      url: Routes.api_v1_website_url($.website_id)
+      type: 'PUT'
+      success: (result)->
+        window.location = "/settings"
+        alert 'Key updated'
+    return
+
+randomString = () ->
```

Comment:
idea nu e sa generezi tu in js keya de api... Avem deja in model o metoda care face asta si care se apeleaza on create. Ar trebuii sa gasesti o solutie sa folosesti tot metoda aia. Metoda ta nu se asigura ca keya asta e unica in baza de date....